### ISSUE TITLE
IO failure observed while running pravega-benchmark  "InvalidMessageException: Unknown wire command: 196608256"

### ISSUE THREAD

**Issue by deenav** [pravega/pravega#3625]
**Problem description**
Observing `io.pravega.shared.protocol.netty.InvalidMessageException: Unknown wire command: 196608256` error in Segmentstore while running IO using pravega-benchmark on  PKS cluster.
`pravega-benchmark` is used to run IO to the controller external-IP on pks cluster.

Deployment Details:
```
Pravega-Operator image: pravega/pravega-operator:0.3.2
Pravega build used : 0.5.0-2205.02f1c00
Zookeeper-operator image: pravega/zookeeper-operator:0.2.1
External-connectivity enabled.

Pravega-benchmark client version: 0.5.0-2066.6ba29eb-SNAPSHOT
```
Environment details: PKS / K8 with large cluster:
```
3 master nodes @ large.cpu (4 CPU, 4 GB Ram, 16 GB Disk)
10 worker nodes @ 2xlarge.cpu(16 cpu, 16 GB Ram, 64 GB Disk)
Tier-1 storage is from VSAN datastore
Tier-2 storage curved on NFS Client provisioner using Isilon as backend 
```

**POD and SVC details:**
```NAMESPACE     NAME                                    TYPE           CLUSTER-IP       EXTERNAL-IP                   PORT(S)                          AGE
default       kubernetes                              ClusterIP      10.100.200.1     <none>                        443/TCP                          37d
default       pravega-bookie-headless                 ClusterIP      None             <none>                        3181/TCP                         40m
default       pravega-pravega-controller              LoadBalancer   10.100.200.37    10.240.119.113,100.64.64.71   10080:30987/TCP,9090:30976/TCP   40m
default       pravega-pravega-segmentstore-0          LoadBalancer   10.100.200.189   10.240.119.108,100.64.64.71   12345:30188/TCP                  40m
default       pravega-pravega-segmentstore-1          LoadBalancer   10.100.200.210   10.240.119.109,100.64.64.71   12345:32255/TCP                  40m
default       pravega-pravega-segmentstore-2          LoadBalancer   10.100.200.26    10.240.119.110,100.64.64.71   12345:30546/TCP                  40m
default       pravega-pravega-segmentstore-3          LoadBalancer   10.100.200.190   10.240.119.111,100.64.64.71   12345:31995/TCP                  40m
default       pravega-pravega-segmentstore-4          LoadBalancer   10.100.200.27    10.240.119.112,100.64.64.71   12345:32653/TCP                  40m
default       pravega-pravega-segmentstore-headless   ClusterIP      None             <none>                        12345/TCP                        40m
```
```
NAMESPACE     NAME                                                 READY   STATUS    RESTARTS   AGE
default       pravega-bookie-0                                     1/1     Running   0          41m
default       pravega-bookie-1                                     1/1     Running   0          41m
default       pravega-bookie-2                                     1/1     Running   0          41m
default       pravega-bookie-3                                     1/1     Running   0          41m
default       pravega-operator-779879b48-2qh6p                     1/1     Running   0          50m
default       pravega-pravega-controller-d5df95fc8-cnr9p           1/1     Running   0          41m
default       pravega-pravega-controller-d5df95fc8-zfxpk           1/1     Running   0          41m
default       pravega-pravega-segmentstore-0                       1/1     Running   0          41m
default       pravega-pravega-segmentstore-1                       1/1     Running   0          40m
default       pravega-pravega-segmentstore-2                       1/1     Running   0          39m
default       pravega-pravega-segmentstore-3                       1/1     Running   0          39m
default       pravega-pravega-segmentstore-4                       1/1     Running   0          38m
```

**SegmentStore-1 log exception**
```
2019-04-12 09:58:31,974 758424 [epollEventLoopGroup-11-7] INFO  i.p.s.s.h.handler.AppendProcessor - Setting up appends for writer: efe9ba84-0902-450e-b2c8-a09c2310b2e0 on segment: Scope/_RGtest/0.#epoch.0
2019-04-12 09:58:32,383 758833 [epollEventLoopGroup-11-7] ERROR i.p.s.s.h.h.ServerConnectionInboundHandler - Caught exception on connection:
io.netty.handler.codec.DecoderException: io.pravega.shared.protocol.netty.InvalidMessageException: Was expecting length: 464 but found: 464
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:472)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
        at io.pravega.shared.protocol.netty.ExceptionLoggingHandler.channelRead(ExceptionLoggingHandler.java:37)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:799)
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:421)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:321)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
Caused by: io.pravega.shared.protocol.netty.InvalidMessageException: Was expecting length: 464 but found: 464
        at io.pravega.shared.protocol.netty.WireCommands$Event.readFrom(WireCommands.java:490)
        at io.pravega.shared.protocol.netty.WireCommands$ConditionalAppend.readFrom(WireCommands.java:639)
        at io.pravega.shared.protocol.netty.WireCommandType.readFrom(WireCommandType.java:122)
        at io.pravega.shared.protocol.netty.CommandDecoder.parseCommand(CommandDecoder.java:55)
        at io.pravega.shared.protocol.netty.CommandDecoder.decode(CommandDecoder.java:36)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441)
        ... 24 common frames omitted
2019-04-12 09:58:32,384 758834 [epollEventLoopGroup-11-7] INFO  i.p.s.s.h.handler.AppendProcessor - Setting up appends for writer: 02000000-002c-0000-0000-0001d4c00000 on segment:
2019-04-12 09:58:32,384 758834 [epollEventLoopGroup-11-7] ERROR i.p.s.s.h.h.ServerConnectionInboundHandler - Caught exception on connection:
io.netty.handler.codec.DecoderException: io.pravega.shared.protocol.netty.InvalidMessageException: Unknown wire command: 196608256
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:472)
        at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:405)
        at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:372)
        at io.netty.handler.codec.ByteToMessageDecoder.channelInactive(ByteToMessageDecoder.java:355)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
        at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:390)
        at io.netty.handler.codec.ByteToMessageDecoder.channelInactive(ByteToMessageDecoder.java:355)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelInactive(ChannelInboundHandlerAdapter.java:75)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1429)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
        at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:947)
        at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:822)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:326)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
Caused by: io.pravega.shared.protocol.netty.InvalidMessageException: Unknown wire command: 196608256
        at io.pravega.shared.protocol.netty.CommandDecoder.readType(CommandDecoder.java:74)
        at io.pravega.shared.protocol.netty.CommandDecoder.parseCommand(CommandDecoder.java:53)
        at io.pravega.shared.protocol.netty.CommandDecoder.decode(CommandDecoder.java:36)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441)
        ... 26 common frames omitted
2019-04-12 09:58:32,386 758836 [core-11] WARN  i.p.s.s.h.handler.AppendProcessor - Segment '' does not exist and 02000000-002c-0000-0000-0001d4c00000 cannot perform operation 'setting up append'.
2019-04-12 09:58:32,386 758836 [epollEventLoopGroup-11-7] WARN  i.n.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.nio.channels.ClosedChannelException: null
        at io.netty.channel.AbstractChannel$AbstractUnsafe.write(...)(Unknown Source)
```

Pravega-benchmark log:
```
[root@master pravega-benchmark]# ./bin/pravega-benchmark -controller tcp://10.240.119.113:9090  --stream test -producers 10 --size 1000  -events 100 --time 259200 -randomkey true -consumers 10 -segments 10
[main] WARN io.pravega.client.ClientConfig - The credentials are not specified or could not be extracted.
[main] WARN io.pravega.client.ClientConfig - The credentials are not specified or could not be extracted.
[main] INFO io.pravega.client.stream.impl.ControllerImpl - Controller client connecting to server at 10.240.119.113:9090
[main] WARN io.pravega.client.ClientConfig - The credentials are not specified or could not be extracted.
[main] WARN io.pravega.client.ClientConfig - The credentials are not specified or could not be extracted.
[main] INFO io.pravega.client.stream.impl.ControllerImpl - Controller client connecting to server at 10.240.119.113:9090
[StreamManager-Controller-1] INFO io.pravega.client.stream.impl.ControllerImpl - [requestId=-8101168555839706464] Tagging client request (createScope-Scope).
[StreamManager-Controller-1] INFO io.pravega.client.stream.impl.ControllerResolverFactory - Updating client with controllers: [[addrs=[/10.240.119.113:9090], attrs={}]]
[grpc-default-executor-0] WARN io.pravega.client.stream.impl.ControllerImpl - [requestId=-8101168555839706464] Scope already exists: Scope
[main] INFO io.pravega.client.admin.impl.StreamManagerImpl - Creating scope/stream: Scope/test with configuration: StreamConfiguration(scope=Scope, streamName=test, scalingPolicy=ScalingPolicy(scaleType=FIXED_NUM_SEGMENTS, targetRate=0, scaleFactor=0, minNumSegments=10), retentionPolicy=null)
[StreamManager-Controller-1] INFO io.pravega.client.stream.impl.ControllerImpl - [requestId=6036870162029010324] Tagging client request (createStream-Scope-test).
[grpc-default-executor-0] INFO io.pravega.client.stream.impl.ControllerImpl - [requestId=6036870162029010324] Stream created successfully: test
[main] WARN io.pravega.client.ClientConfig - The credentials are not specified or could not be extracted.
[main] WARN io.pravega.client.ClientConfig - The credentials are not specified or could not be extracted.
[main] WARN io.pravega.client.ClientConfig - The credentials are not specified or could not be extracted.
[main] INFO io.pravega.client.stream.impl.ControllerImpl - Controller client connecting to server at 10.240.119.113:9090
[main] INFO io.pravega.client.admin.impl.ReaderGroupManagerImpl - Creating reader group: test for streams: [StreamImpl(scope=Scope, streamName=test)] with configuration: ReaderGroupConfig(groupRefreshTimeMillis=3000, automaticCheckpointIntervalMillis=120000, startingStreamCuts={StreamImpl(scope=Scope, streamName=test)=UNBOUNDED}, endingStreamCuts={StreamImpl(scope=Scope, streamName=test)=UNBOUNDED}, maxOutstandingCheckpointRequest=3)
[clientInternal-1] INFO io.pravega.client.stream.impl.ControllerImpl - [requestId=8571397408842266747] Tagging client request (createStream-Scope-_RGtest).
[clientInternal-1] INFO io.pravega.client.stream.impl.ControllerResolverFactory - Updating client with controllers: [[addrs=[/10.240.119.113:9090], attrs={}]]
[grpc-default-executor-0] INFO io.pravega.client.stream.impl.ControllerImpl - [requestId=8571397408842266747] Stream created successfully: _RGtest
[main] INFO io.pravega.client.stream.impl.ClientFactoryImpl - Creating state synchronizer with stream: _RGtest and configuration: SynchronizerConfig(eventWriterConfig=EventWriterConfig(initalBackoffMillis=1, maxBackoffMillis=20000, retryAttempts=10, backoffMultiple=10, transactionTimeoutTime=29999))
[main] INFO io.pravega.client.stream.impl.ClientFactoryImpl - Creating revisioned stream client for stream: _RGtest with synchronizer configuration: SynchronizerConfig(eventWriterConfig=EventWriterConfig(initalBackoffMillis=1, maxBackoffMillis=20000, retryAttempts=10, backoffMultiple=10, transactionTimeoutTime=29999))
[clientInternal-3] INFO io.pravega.client.segment.impl.SegmentOutputStreamImpl - Fetching endpoint for segment Scope/_RGtest/0.#epoch.0, writerID: ffa9c276-d96e-4463-be51-5db98d65c354
[clientInternal-4] INFO io.pravega.client.segment.impl.SegmentOutputStreamImpl - Establishing connection to PravegaNodeUri(endpoint=10.240.119.109, port=12345) for Scope/_RGtest/0.#epoch.0, writerID: ffa9c276-d96e-4463-be51-5db98d65c354
[epollEventLoopGroup-6-1] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0xdd2d5ea0])
[epollEventLoopGroup-6-2] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0x79a0610c])
[epollEventLoopGroup-6-3] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0x9e37795d])
[epollEventLoopGroup-6-1] INFO io.pravega.shared.protocol.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-3] INFO io.pravega.client.netty.impl.RawClient - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-2] INFO io.pravega.shared.protocol.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-2] INFO io.pravega.client.segment.impl.SegmentOutputStreamImpl - Received AppendSetup WireCommands.AppendSetup(type=APPEND_SETUP, requestId=1, segment=Scope/_RGtest/0.#epoch.0, writerId=ffa9c276-d96e-4463-be51-5db98d65c354, lastEventNumber=-9223372036854775808)
[epollEventLoopGroup-6-2] INFO io.pravega.client.segment.impl.SegmentOutputStreamImpl - Connection setup complete for writer ffa9c276-d96e-4463-be51-5db98d65c354
[epollEventLoopGroup-6-4] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0xc55be941])
[epollEventLoopGroup-6-4] INFO io.pravega.client.netty.impl.RawClient - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-4] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[epollEventLoopGroup-6-5] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0x40485e92])
[epollEventLoopGroup-6-5] INFO io.pravega.client.netty.impl.RawClient - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-5] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[epollEventLoopGroup-6-6] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0x87e69b86])
[epollEventLoopGroup-6-6] INFO io.pravega.client.netty.impl.RawClient - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-6] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[epollEventLoopGroup-6-7] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0x5a96aa5f])
[epollEventLoopGroup-6-7] INFO io.pravega.client.netty.impl.RawClient - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-7] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[epollEventLoopGroup-6-8] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0xc7ccf9d3])
[epollEventLoopGroup-6-8] INFO io.pravega.client.netty.impl.RawClient - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-8] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null

[epollEventLoopGroup-6-1] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0x42f79265])
[epollEventLoopGroup-6-1] INFO io.pravega.client.netty.impl.RawClient - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-1] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[epollEventLoopGroup-6-2] INFO io.pravega.client.netty.impl.ClientConnectionInboundHandler - Connection established ChannelHandlerContext(ClientConnectionInboundHandler#0, [id: 0x4ed9becc])
[epollEventLoopGroup-6-2] INFO io.pravega.client.netty.impl.RawClient - Received hello: WireCommands.Hello(type=HELLO, highVersion=8, lowVersion=5)
[epollEventLoopGroup-6-2] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[main] INFO io.pravega.client.state.impl.StateSynchronizerImpl - Closing stateSynchronizer
[main] INFO io.pravega.client.segment.impl.ConditionalOutputStreamImpl - Closing segment metadata connection for Scope/_RGtest/0.#epoch.0
[main] INFO io.pravega.client.segment.impl.ConditionalOutputStreamImpl - Closing connection as a result of receiving: Closed call
[epollEventLoopGroup-6-2] INFO io.pravega.client.segment.impl.SegmentOutputStreamImpl - Failing connection for writer ffa9c276-d96e-4463-be51-5db98d65c354 with exception io.pravega.shared.protocol.netty.ConnectionFailedException
[main] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[main] INFO io.pravega.client.segment.impl.SegmentMetadataClientImpl - Closing segment metadata connection for Scope/_RGtest/0.#epoch.0
[epollEventLoopGroup-6-2] INFO io.pravega.client.segment.impl.SegmentOutputStreamImpl - Handling exception io.pravega.shared.protocol.netty.ConnectionFailedException for connection io.pravega.client.netty.impl.ClientConnectionInboundHandler@7814d290 on writer ffa9c276-d96e-4463-be51-5db98d65c354. SetupCompleted: true, Closed: true
[main] INFO io.pravega.client.segment.impl.SegmentMetadataClientImpl - Closing connection with exception: null
[main] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[main] INFO io.pravega.client.segment.impl.AsyncSegmentInputStreamImpl - Closing reader for Scope/_RGtest/0.#epoch.0
[epollEventLoopGroup-6-3] INFO io.pravega.client.netty.impl.RawClient - Closing connection with exception: null
[main] INFO io.pravega.client.segment.impl.AsyncSegmentInputStreamImpl - Closing connection to segment: Scope/_RGtest/0.#epoch.0
[main] INFO io.pravega.client.segment.impl.AsyncSegmentInputStreamImpl - Connection failed due to a io.pravega.client.stream.impl.ConnectionClosedException. Read requests will be retransmitted.
[epollEventLoopGroup-6-1] INFO io.pravega.client.segment.impl.AsyncSegmentInputStreamImpl - Closing connection to segment: Scope/_RGtest/0.#epoch.0
[epollEventLoopGroup-6-1] INFO io.pravega.client.segment.impl.AsyncSegmentInputStreamImpl - Connection failed due to a io.pravega.shared.protocol.netty.ConnectionFailedException. Read requests will be retransmitted.
[main] WARN io.pravega.client.segment.impl.AsyncSegmentInputStreamImpl - Exception while reading from Segment : Scope/_RGtest/0.#epoch.0
io.pravega.client.stream.impl.ConnectionClosedException
        at io.pravega.client.segment.impl.AsyncSegmentInputStreamImpl.close(AsyncSegmentInputStreamImpl.java:149)
        at io.pravega.client.segment.impl.SegmentInputStreamImpl.close(SegmentInputStreamImpl.java:203)
        at io.pravega.client.segment.impl.EventSegmentReaderImpl.close(EventSegmentReaderImpl.java:109)
        at io.pravega.client.state.impl.RevisionedStreamClientImpl.close(RevisionedStreamClientImpl.java:201)
        at io.pravega.client.state.impl.StateSynchronizerImpl.close(StateSynchronizerImpl.java:251)
        at io.pravega.client.admin.impl.ReaderGroupManagerImpl.createReaderGroup(ReaderGroupManagerImpl.java:98)
        at com.emc.pravega.perf.PravegaStreamHandler.createReaderGroup(PravegaStreamHandler.java:149)
        at com.emc.pravega.perf.PravegaPerfTest.main(PravegaPerfTest.java:124)
io.pravega.common.util.RetriesExhaustedException: java.util.concurrent.CompletionException: io.pravega.shared.protocol.netty.ConnectionFailedException
        at io.pravega.common.util.Retry$RetryAndThrowBase.run(Retry.java:193)
        at io.pravega.client.segment.impl.ConditionalOutputStreamImpl.write(ConditionalOutputStreamImpl.java:68)
        at io.pravega.client.state.impl.RevisionedStreamClientImpl.writeConditionally(RevisionedStreamClientImpl.java:67)
        at io.pravega.client.state.impl.StateSynchronizerImpl.initialize(StateSynchronizerImpl.java:167)
        at io.pravega.client.admin.impl.ReaderGroupManagerImpl.createReaderGroup(ReaderGroupManagerImpl.java:102)
        at com.emc.pravega.perf.PravegaStreamHandler.createReaderGroup(PravegaStreamHandler.java:149)
        at com.emc.pravega.perf.PravegaPerfTest.main(PravegaPerfTest.java:124)
Caused by: java.util.concurrent.CompletionException: io.pravega.shared.protocol.netty.ConnectionFailedException
        at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
        at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
        at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:911)
        at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:899)
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
        at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
        at io.pravega.client.netty.impl.RawClient.closeConnection(RawClient.java:111)
        at io.pravega.client.netty.impl.RawClient.access$200(RawClient.java:35)
        at io.pravega.client.netty.impl.RawClient$ResponseProcessor.connectionDropped(RawClient.java:62)
        at io.pravega.client.netty.impl.ClientConnectionInboundHandler.channelUnregistered(ClientConnectionInboundHandler.java:87)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:181)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:167)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelUnregistered(AbstractChannelHandlerContext.java:160)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelUnregistered(ChannelInboundHandlerAdapter.java:53)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:181)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:167)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelUnregistered(AbstractChannelHandlerContext.java:160)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelUnregistered(ChannelInboundHandlerAdapter.java:53)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:181)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:167)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelUnregistered(AbstractChannelHandlerContext.java:160)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelUnregistered(ChannelInboundHandlerAdapter.java:53)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:181)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:167)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelUnregistered(AbstractChannelHandlerContext.java:160)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelUnregistered(DefaultChannelPipeline.java:1337)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:181)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelUnregistered(AbstractChannelHandlerContext.java:167)
        at io.netty.channel.DefaultChannelPipeline.fireChannelUnregistered(DefaultChannelPipeline.java:835)
        at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:830)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:309)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
        at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
        at java.lang.Thread.run(Thread.java:748)
Caused by: io.pravega.shared.protocol.netty.ConnectionFailedException
        ... 28 more

```

**Comment by deenav** [comment]
Attaching controller & segmentstore logs 
[pravega-controller-logs.zip](https://github.com/pravega/pravega/files/3072963/pravega-controller-logs.zip)
[segmentstore-logs.zip](https://github.com/pravega/pravega/files/3072964/segmentstore-logs.zip)

**Comment by deenav** [comment]
Modified the issue title since this issue is not related to `external-connectivity`. 

This issue is due to the Pravega-benchmark with older Pravega client version `0.5.0-2066.6ba29eb-SNAPSHOT` is not compatible & throwing below errors with latest Pravega server version `0.5.0-2205.02f1c00`.

```
2019-04-12 09:58:32,383 758833 [epollEventLoopGroup-11-7] ERROR i.p.s.s.h.h.ServerConnectionInboundHandler - Caught exception on connection:
io.netty.handler.codec.DecoderException: io.pravega.shared.protocol.netty.InvalidMessageException: Was expecting length: 464 but found: 464
```

After rebuilding Pravega-benchmark with latest pravega client version, IO's are running fine on external-connectivity enabled PKS cluster. 

I see that issue #3537 addresses "Backward Compatability of WireCommands.ConditionalAppend" which will resolve this issue.

**Comment by RaulGracia** [comment]
Potential duplicate of #3624.

**Comment by RaulGracia** [comment]
This issue should be solved via #3640 as for #3624. If the issue persists, please re-open it.
